src/app.ts

```diff
@@ -180,9 +180,14 @@ export const initializeAppModules = () => {
       inputs: [{
         type: "boolean",
         name: "mockOAuth" // do not go to the actual oauth provider, for scenario testing purpose
+      },
+      {
+        type: "string",
+        name: "hubAuthCallbackUrl"
       }],
       hasOutput: true,
-      action: function (mockOAuth, output) {
+      action: function (mockOAuth, output, hubAuthCallbackUrl) {
+        localStorage.setItem("hubAuthCallbackUrl", encodeURIComponent(hubAuthCallbackUrl))

```

Comment:
@EFFYLYX, I don't think this change will work as intended. The code in app.ts runs inside of the MetaOS Hub whereas the auth_end.html callback runs inside the browser. The localStorage is not shared between these two so anything you write here will not be available in the auth_end.html callback where you are trying to read it.